joern-cli/frontends/jssrc2cpg/src/main/scala/io/joern/jssrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -159,6 +161,63 @@ trait AstForDeclarationsCreator { this: AstCreator =>
     }
   }
 
+  protected def astsForDecorators(elem: BabelNodeInfo): Seq[Ast] = {
+    if (hasKey(elem.json, "decorators") && !elem.json("decorators").isNull) {
+      elem.json("decorators").arr.toList.map(d => astForDecorator(createBabelNodeInfo(d)))
+    } else Seq.empty
+  }
+
+  private def createAnnotationAssignmentAst(name: String, value: Value, code: String): Ast = {
```

Comment:
This one looks like we may want to split it up into a part that depends on the `jssrc2cpg` specific parse tree and the part that deals with the creation of the CPG-AST. We could then make it available via `AstCreatorBase`. Could you add a comment about that if you agree?

Reply:
Yes, makes sense. Do you want to do this right away with this PR here or as a separate refactoring?

Reply:
Feel free to merge.